modules/input.nix

```diff
@@ -0,0 +1,386 @@
+      enable = mkOption {
+        type = with types; nullOr bool;
+        default = null;
+        example = true;
+        description = ''
+          Enables or disables the mouse
+        '';
+      };
+      name = mkOption {
+        type = types.str;
+        default = null;
+        example = "Logitech G403 HERO Gaming Mouse";
+        description = ''
+          The name of the mouse.
+          You can find it out running cat /proc/bus/input/devices | grep -B 1 -i mouse
+        '';
+      };
+      vendorId = mkOption {
+        type = types.str;
+        default = null;
+        example = "046d";
+        description = ''
+          The vendor ID of the mouse.
+          You can find it out running cat /proc/bus/input/devices | grep -B 1 -i mouse
+        '';
+      };
+      productId = mkOption {
+        type = types.str;
+        default = null;
+        example = "c077";
+        description = ''
+          The product ID of the mouse.
+          You can find it out running cat /proc/bus/input/devices | grep -B 1 -i mouse
+        '';
+      };
+      leftHanded = mkOption {
+        type = with types; nullOr bool;
+        default = null;
+        example = false;
+        description = ''
+          Swap the left and right buttons
+        '';
+      };
+      middleMouseEmulation = mkOption {
+        type = with types; nullOr bool;
+        default = null;
+        example = false;
+        description = ''
+          Middle click by pressing the left and right buttons at the same time.
+          Activating this increases the click latency by 50ms
+        '';
+      };
+      acceleration = mkOption {
+        type = with types; nullOr (numbers.between (-1) 1);
+        default = null;
+        example = 0.5;
+        description = ''
+          Mouse acceleration
+        '';
+      };
+      accelerationProfile = mkOption {
+        type = with types; nullOr (enum [ "none" "default" ]);
+        default = null;
+        example = "none";
+        description = "Mouse acceleration profile";
+        apply = profile: if profile == "none" then 1 else if profile == "default" then 2 else null;
+      };
+      naturalScroll = mkOption {
+        type = with types; nullOr bool;
+        default = null;
+        example = true;
+        description = ''
+          Enables natural scrolling for the mouse.
+        '';
+      };
+      scrollSpeed = mkOption {
+        type = with types; nullOr (numbers.between 0.1 20);
+        default = null;
+        example = 1;
+        description = ''
+          How fast the scroll wheel moves
+        '';
+      };
+    };
+  };
+
+  mouseToConfig = mouse:
+    let
+      mouseName = mouse.name;
+      mouseVendor = mouse.vendorId;
+      mouseProduct = mouse.productId;
+    in
+    {
+      "Libinput/${mouseVendor}/${mouseProduct}/${mouseName}" = {
+        Enabled = mouse.enable;
+        LeftHanded = mouse.leftHanded;
+        MiddleMouseEmulation = mouse.middleMouseEmulation;
```

Comment:
I also think it should be `MiddleButtonEmulation` here

Reply:
@toast003 could you fix this?

Reply:
Done

Reply:
Thanks!